src/OSDHistory.c

```diff
@@ -126,40 +126,43 @@ static u16 GetTimestamp(void)
 int AddHistoryRecord(const char *name)
 {
     struct HistoryEntry HistoryEntries[MAX_HISTORY_ENTRIES], *NewEntry, OldHistoryEntry;
-    int i, value, LeastUsedRecord, LeastUsedRecordLaunchCount, LeastUsedRecordTimestamp, NewLaunchCount, result, mcType;
+    int i, value, LeastUsedRecord, LeastUsedRecordLaunchCount, LeastUsedRecordTimestamp, NewLaunchCount, result, mcType, format;
     u8 BlankSlotList[MAX_HISTORY_ENTRIES];
     int NumBlankSlots, NumSlotsUsed, IsNewRecord;
     char SystemRegionLetter;
     char path[32];
+    DEBUG_PRINTF("OSDHistory@%s: starts\n", __func__);
 
-    DEBUG_PRINTF("Adding history record: %s\n", name);
+    for (i = 0; i < 2; i++) { //increase number of slots for mt (not sure how multi tap works haven’t looked at docs ie mc0 - mc4 ?)
+        mcGetInfo(i, 0, &mcType, NULL, &format);
+        mcSync(0, NULL, &result);
+        DEBUG_PRINTF("\tslot=%d, mctype=%d, format=%d\n", i, mcType, format);
+        if ((mcType == sceMcTypePS2) && (format == MC_FORMATTED))
+            break;
+    }
 
-    // Don't write history for ps1 cards
-    mcGetInfo(0, 0, &mcType, 0, 0);
-    mcSync(0, NULL, &result);
-    if (mcType == sceMcTypePS1)
+    if ((mcType == sceMcTypePS2) && (format == MC_FORMATTED)) { // don't even waste time if there are no PS2 MC's
```

Comment:
this check should be `!=` mate, purpose being to determine whether the loop above ended because a valid card was found or if it just ran all iterations without finding a valid card.. few different ways to do this but this seemed quick and easy

Reply:
Yeah, sorry.

Made it on cellphone and missed that one